kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/filter/oauthbearer/OauthBearerValidationIT.java

```diff
@@ -0,0 +1,282 @@
+                    .isPositive();
+
+            assertThat(saslAuthenticationResponsesComingDown).isNotPresent();
+        }
+        finally {
+            @SuppressWarnings("java:S1481")
+            var ignored = badTokenFile.toFile().delete();
+        }
+
+    }
+
+    private Optional<Double> findFirstMetricMatching(List<SimpleMetric> all, Predicate<SimpleMetric> predicate) {
+        return all.stream()
+                .filter(predicate)
+                .findFirst()
+                .map(SimpleMetric::value);
+    }
+
+    private ConfigurationBuilder getConfiguredProxyBuilder() {
+        return proxy(cluster)
+                .withNewAdminHttp()
+                .withNewEndpoints()
+                .withNewPrometheus()
+                .endPrometheus()
+                .endEndpoints()
+                .endAdminHttp()
+                .addToFilters(new FilterDefinitionBuilder(
+                        OauthBearerValidation.class.getName())
+                        .withConfig("jwksEndpointUrl", JWKS_ENDPOINT_URL,
+                                "expectedAudience", EXPECTED_AUDIENCE)
+                        .build());
+    }
+
+    @NonNull
+    private Map<String, Object> getClientConfig(URI tokenEndpointUrl) {
+        var oauthClientConfig = new HashMap<String, Object>();
+        oauthClientConfig.put(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG, SecurityProtocol.SASL_PLAINTEXT.name());
+        oauthClientConfig.put(SaslConfigs.SASL_MECHANISM, OAuthBearerLoginModule.OAUTHBEARER_MECHANISM);
+        oauthClientConfig.put(SaslConfigs.SASL_OAUTHBEARER_TOKEN_ENDPOINT_URL, tokenEndpointUrl.toString());
+        oauthClientConfig.put(SaslConfigs.SASL_LOGIN_CALLBACK_HANDLER_CLASS, OAuthBearerLoginCallbackHandler.class.getCanonicalName());
+        oauthClientConfig.put(SaslConfigs.SASL_JAAS_CONFIG,
+                "%s required clientId=\"clientIdIgnore\" clientSecret=\"clientSecretIgnore\";".formatted(OAuthBearerLoginModule.class.getName()));
+        return oauthClientConfig;
+    }
+
+    /**
+     * Pings the cluster in order to assert connectivity. We don't care about the result.
+     * @param admin admin
+     */
+    @SuppressWarnings("java:S1481") // making clear the intent that the result of the operation is unneeded.
+    private void performClusterOperation(Admin admin) {
+        var unused = admin.describeCluster().nodes().toCompletionStage().toCompletableFuture().join();
+        assertThat(unused).isNotNull();
+    }
+
+    private static class OauthServerContainer extends GenericContainer<OauthServerContainer> {
+        private OauthServerContainer(DockerImageName dockerImageName) {
+            super(dockerImageName);
+        }
+
+        protected void addFixedExposedPort(int hostPort, int containerPort) {
+            super.addFixedExposedPort(hostPort, containerPort);
+        }
+    }
+
+    private record SimpleMetric(String name, Map<String, String> labels, double value) {
```

Comment:
We might want to move something like this to the Tester.

Reply:
Yeah I was saying something similar at the call site.

Reply:
I'll move it across to Tester as a separate PR.